### PR TITLE
Fix torch.mm export to ONNX

### PR DESCRIPTION
**Pull request by KsenijaS**
torch.mm is exported as Gemm operator in ONNX and both have an optional input: out.
out is considered as broadcastable in Gemm and during graph optimization the optional input (out) would get selected. Since out is optional, in case when it is not defined in torch.mm that would result in the following exception:
IndexError: vector::_M_range_check: __n (which is 2) >= this->size() (which is 2)

### CODE REVIEW

**Comment by BowenBao** [comment]
Please add the failing repro as a test case.

File: torch/csrc/jit/passes/onnx/peephole.cpp [hunk 1]
```diff
@@ -109,6 +109,9 @@ void fuseBroadcast(Block* b) {
     }
 
     for (size_t position : broadcast_positions) {
+      if (position >= n->inputs().size()) {
```
Comment (BowenBao) [inline torch/csrc/jit/passes/onnx/peephole.cpp]: In general I would recommend asserting that `position` is valid. It should be expected that `getBroadcastPositions` does not return invalid results. So the fix to skip optional input for `onnx::Gemm`(as a matter of fact, any operator with optinal inputs) should go into `getBroadcastPositions`.

**Review by neginraoof — APPROVED** [review]
LGTM

**Comment by KsenijaS** [comment]
@houseroad can you please review?

**Review by houseroad — APPROVED** [review]
Looks good

File: torch/csrc/jit/passes/onnx/peephole.cpp [hunk 2]
```diff
@@ -68,9 +68,16 @@ const std::vector<size_t>& getBroadcastPositions(Node* node) {
           {onnx::Less, {0, 1}},
       };
   static std::vector<size_t> no_positions;
+  static std::vector<size_t> positions;
 
   auto iter = broadcast_positions.find(node->kind());
   if (iter != broadcast_positions.end()) {
+    for (long unsigned int i = 0; i < iter->second.size(); i++) {
```
Comment (BowenBao) [inline torch/csrc/jit/passes/onnx/peephole.cpp]: Synced offline with Ksenija, some minor changes will be added here.

**Comment by neginraoof** [comment]
@KsenijaS  Please check ONNX test failures.

File: torch/csrc/jit/passes/onnx/peephole.cpp [hunk 3]
```diff
@@ -113,7 +120,7 @@ void fuseBroadcast(Block* b) {
 
       // Confirm it is expand node.
       if (expand_node->kind() != aten::expand ||
-          expand_node->input(1)->node()->kind() != onnx::Constant ||
+
```
Comment (BowenBao) [inline torch/csrc/jit/passes/onnx/peephole.cpp]: nit: typo?

**Comment by KsenijaS** [comment]
@houseroad Can you please take another look at this PR? The CI failure is not related to my changes.

**Review by BowenBao — APPROVED** [review]

**Pull request closed without merging**